### PR TITLE
[OASIS-37] Move collector internal/datadog to Agent as modules

### PR DESCRIPTION
**Pull request by songy23**
### What does this PR do?

Copy https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/internal/datadog to Agent and export them as Go modules.

Will cleanup https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/internal/datadog and have Datadog connector depend on these modules next.

### Motivation

Attempt to resolve the circular dependency between collector and agent.

### Describe how to test/QA your changes

N/A this has no user-facing impact to Agent users

### CODE REVIEW

**Review by songy23 — COMMENTED** [review]
All the Go codes are copy-paste of https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/internal/datadog

File: comp/otelcol/otlp/components/ingester/agent.go [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+
+package ingester
+
+import (
+	"context"
+	"net/http"
+	"runtime"
+	"sync"
+	"time"
+
```
Comment (dineshg13) [inline comp/otelcol/otlp/components/ingester/agent.go:12]: can we call it stats processor ?

**Review by FlorentClarret — APPROVED** [review]
👍 for ADT

**Review by CelianR — APPROVED** [review]

File: comp/otelcol/otlp/components/statsprocessor/agent.go [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+)
+
+// TraceAgent specifies a minimal trace agent instance that is able to process traces and output stats.
+type TraceAgent struct {
+	*agent.Agent
+
+	// pchan specifies the channel that will be used to output Datadog Trace Agent API Payloads
+	// resulting from ingested OpenTelemetry spans.
+	pchan chan *api.Payload
+
```
Comment (dineshg13) [inline comp/otelcol/otlp/components/statsprocessor/agent.go:31]: Can we make stats processor a component ? We can we have two implementations one with `pkg/trace` and other with new interface we added for stats API ?

**Review by dineshg13 — CHANGES_REQUESTED** [review]

File: comp/otelcol/otlp/components/statsprocessor/agent.go (hunk 2 above)
Comment (songy23) [reply comp/otelcol/otlp/components/statsprocessor/agent.go:31]: That can be in a follow-up PR. Also it won't be of any use to upstream Datadog connector.

**Review by dineshg13 — APPROVED** [review]

**Pull request merged**